src/script/team/TeamRepository.ts

```diff
@@ -427,13 +432,46 @@ export class TeamRepository {
       this.handleConferenceCallingFeatureChange(previousConfig, featureConfigList);
       this.handleGuestLinkFeatureChange(previousConfig, featureConfigList);
     }
+
+    // Trigger E2E identity handler
+    this.handleE2EIdentityFeatureChange(featureConfigList.mlsE2EId, featureConfigList.mls);
+
     this.saveFeatureConfig(featureConfigList);
   };
 
+  private readonly handleE2EIdentityFeatureChange = (mlsE2EId?: FeatureMLSE2EId, mls?: FeatureMLS) => {
+    if (!mlsE2EId || !mls) {
+      return;
+    }
+
+    const isE2EIActive = mlsE2EId?.status === FeatureStatus.ENABLED;
+    const isMLSActive = mls?.status === FeatureStatus.ENABLED;
+    const hasConfig =
+      mlsE2EId?.config && mlsE2EId?.config.acmeDiscoveryUrl && mlsE2EId?.config.acmeDiscoveryUrl.length > 0;
+
+    if (isE2EIActive) {
+      if (!isMLSActive) {
+        this.logger.info('Warning: E2E identity feature enabled but MLS feature is not active');
```

Comment:
how about adding small functions for each of the if statement here?
checkIsMLSActive();
checkHasConfig();

it will be more readable

Reply:
This would only be benefitial if we share these functions or reuse them in several places, I think